pkg/channel/consolidated/dispatcher/dispatcher.go

```diff
@@ -248,9 +324,17 @@ func (d *KafkaDispatcher) UpdateKafkaConsumers(config *Config) (map[types.UID]er
 
 			// Check if sub already exists
 			exists := false
-			for _, s := range d.channelSubscriptions[channelRef] {
-				if s == subSpec.UID {
-					exists = true
+			if _, ok := d.channelSubscriptions[channelRef]; ok {
+				for _, s := range d.channelSubscriptions[channelRef].subs {
+					if s == subSpec.UID {
+						exists = true
+					}
+				}
+			} else { //ensure the pointer is populated or things go boom
+				d.channelSubscriptions[channelRef] = &KafkaSubscription{
```

Comment:
I've a feeling this needs to be protected with a lock, wdyt?

Reply:
You mean like the outer lock `d.consumerUpdateLock.Lock()` on L236?